ott/core/bar_problems.py

```diff
@@ -152,6 +164,7 @@ def max_measure_size(self) -> int:
 
   @property
   def num_segments(self) -> int:
+    """Number of measures."""
     if self._y is None:
```

Comment:
I think you defined this to be no longer optional?

Reply:
or maybe this is just there to help with the jit "dry run" compilation?

Reply:
> I think you defined this to be no longer optional?

You're right. I've refactored the code after consulting with @ersisimou so that optionally segmentation happens in `__init__`. Tests locally passed, so hope it didn't break anything else.